### PR TITLE
linux_6_1,linux_6_5: backport patch from 6.6 for NFS fix

### LINKED ISSUE #255803: Mounting NFS volumes stopped working
### Describe the bug
Mounting NFS volumes stopped working without configuration change other than updates.

### Steps To Reproduce
1. Get updated system as of 2023-09-18 from stable
2. Try to mount a nfs volume
3.  Get the error
```
sudo mount.nfs 192.168.7.2:/volume/xxx/xxx /mnt/test
Failed to start rpc-statd.service: Unit rpc-statd.service not found.
mount.nfs: rpc.statd is not running but is required for remote locking.
mount.nfs: Either use '-o nolock' to keep locks local, or start statd.
mount.nfs: an incorrect mount option was specified

```

### Expected behavior
The volume to be mounted

### Notify maintainers

@abbradar 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.5.3, NixOS, 23.05 (Stoat), 23.05.3580.5d017a8822e0`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.13.5`
 - channels(root): `"home-manager-23.05.tar.gz, nixos-23.05"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

```

### PR DESCRIPTION
**Pull request by nevivurn**
## Description of changes

fsid=0 is special, and due to its special treatment the tests failed to catch #255803, introduced during kernel updates 6.1.52->6.1.53 and 6.5.2->6.5.3. With this change, the tests correctly fail without the backported patch.

This issue currently persists on 6.1.54, and to fix the actual issue, we need to pull in "selinux: fix handling of empty opts in selinux_fs_context_submount()" (patch [here](https://git.kernel.org/pub/scm/linux/kernel/git/pcmoore/selinux.git/commit/?h=stable-6.6&id=ccf1dab96be4caed7c5235b1cfdb606ac161b996), see the Gentoo folks' work [here](https://bugs.gentoo.org/914204) and Fedora [here](https://bugzilla.redhat.com/show_bug.cgi?id=2236345)).

Tested:
- Updated tests fail on 6.1.52, unpatched 6.1.54, unpatched 6.5.4.
- Patch applies cleanly & builds linux_6_1 and linux_6_5.
- Updated tests succeed on 6.1.52, patched 6.1.54, patched 6.5.4.

Fixes #255803.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/tests/nfs/kerberos.nix [hunk 1]
```diff
@@ -71,7 +71,7 @@ in
         services.nfs.server.createMountPoints = true;
         services.nfs.server.exports =
           ''
-            /data *(rw,no_root_squash,fsid=0,sec=krb5p)
+            /data *(rw,no_root_squash,fsid=1,sec=krb5p)
```
Comment (Mic92) [inline nixos/tests/nfs/kerberos.nix:74]: Interesting. I was using an fsid: https://github.com/TUM-DSE/doctor-cluster-config/blob/e88b4edcdbeb0701a4606783cce083e8e9572fde/modules/nfs/server.nix#L41 but still saw breakages.

Comment (Mic92) [reply nixos/tests/nfs/kerberos.nix:74]: Ah. It's actually not using fsid=0 that breaks it?

Comment (nevivurn) [reply nixos/tests/nfs/kerberos.nix:74]: Yeah, this "fix" is making the test fail. It currently passes even though NFS4 is broken for many users.

**Comment by alyssais** [comment]
Is it queued for a stable backport upstream?

**Comment by nevivurn** [comment]
> Is it queued for a stable backport upstream?

I am not familiar with the upstream kernel backport process, but the upstream selinux maintainer [said](https://lore.kernel.org/selinux/f48a346737f99d7b82ecaf214ac2b77e.paul@paul-moore.com/):

> I'm also tagging this for the stable kernels even though this patch
> is only present in v6.6-rc1 because the original patch has a number
> of 'Fixes:' tags which means the stable folks will probably end up
> pulling it into their trees.

So it's supposed to eventually land on stable, I think?

File: pkgs/os-specific/linux/kernel/patches.nix [hunk 2]
```diff
@@ -34,6 +34,11 @@
       patch = ./modinst-arg-list-too-long.patch;
     };
 
+  selinux_empty_opts =
+    { name = "selinux-empty-opts";
+      patch = ./selinux-empty-opts.patch;
```
Comment (alyssais) [inline pkgs/os-specific/linux/kernel/patches.nix:39]: Why can't we fetchurl https://lore.kernel.org/selinux/20230911142358.883728-1-omosnace@redhat.com/raw?

Comment (nevivurn) [reply pkgs/os-specific/linux/kernel/patches.nix]: Done!

**Review by alyssais — APPROVED** [review]
Kernel changes look fine to me.

**Review by adamcstephens — APPROVED** [review]
This fixes the problem for me on 6.1. (I did not test 6.5)

**Comment by adamcstephens** [comment]
The patch is in 6.1.55 and 6.5.5

**Comment by adamcstephens** [comment]
Are the changes to tests here still desirable?

**Comment by Mic92** [comment]
Potentially?

**Comment by Artturin** [comment]
https://github.com/NixOS/nixpkgs/pull/257663

**Pull request closed without merging**